### PR TITLE
Use web3icons

### LINKED ISSUE #117: Extra chains logos
### Description: 

In #109 we allowed users to add more chains(chains which are provided by wagmi) to dropdown, Maybe we could denote somehow that this chains were added by user? Maybe different logo or some extra text?  Currently we show same logo(Ethereum mainnet logo) for all the added chains. 
image

### PR DESCRIPTION
**Pull request by 0xa3k5**
## Description

uses `NetworkIcon` from [web3icons](https://tokenicons.io) in the NetworksDropdown component. The NetworkIcon component relies on a chain's id (`chainId`) so it is future proof, most of the chains in the `viem` is supported in the library, and library is growing.

I've specifically did not delete the .svg images under the public folder, from what I can tell, they are used in OG image generation, I'm not 100% sure if there are any other place that uses the `.icon` properties that relies on the `.svg` files. Feel free to expand on the PR to include them if you think it's necessary.

https://share.cleanshot.com/G3p8vj1s

## Additional Information

- [x] I have read the [contributing docs](https://github.com/BuidlGuidl/abi.ninja/blob/main/CONTRIBUTING.md) (if this is your first contribution)
- [x] This is not a duplicate of any [existing pull request](https://github.com/BuidlGuidl/abi.ninja/pulls)

## Related Issues

Closes #117 

Your ENS/address: a3k5.eth

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -740,6 +740,173 @@ __metadata:
...
+  linkType: hard
+
+"@inquirer/type@npm:^1.5.2":
+  version: 1.5.2
+  resolution: "@inquirer/type@npm:1.5.2"
+  dependencies:
+    mute-stream: ^1.0.0
+  checksum: 6f5ee8143c982aaab0d14871f346b5770c9630e0f14303119c03fb9cb7904f6e0af0b76dcef9528da16563877f65e80a6056aa6d90e77511ab42177cdf06c4d7
+  languageName: node
+  linkType: hard
```
Comment (technophile-04) [inline yarn.lock:908]: Just curious why we need `inquirer` it seems to useful only when building cli applications

**Review by technophile-04 — APPROVED** [review]
Looks good to me! The only concern for me is the library size

image

But makes sense to have it large since we have svg images in their. Also once advantage is that when people add chain from "other chain"  they get the image out of box. Also I don't see any delay in website initial load (maybe next optimising thing at build time)

**Comment by 0xa3k5** [comment]
@technophile-04 the `inquirer` is used in the `@web3icons/utils` which is a direct dependency of `@web3icons/react` I'm working on some improvements which should help reduce the unpacked size and also reduce the dependencies of the package. Thanks for the review, I really like abi.ninja, joined the Discord server, lmk if I can contribute any other way

**Review by portdeveloper — APPROVED** [review]
There is a minor layout shift but that could be because of my connection!!
we can fix that later if it becomes a problem

lgtm!

**Comment by 0xa3k5** [comment]
hey @technophile-04 & @portdeveloper I've just released a new version to the @web3icons/react that reduces it's dependencies (mainly inquirer) as well as introduces some more network icons.

**Comment by portdeveloper** [comment]
> hey @technophile-04 & @portdeveloper I've just released a new version to the @web3icons/react that reduces it's dependencies (mainly inquirer) as well as introduces some more network icons, thought I'd commit to the PR while it's still open.

Thank you for the hard work! I think what you're doing with web3icons is amazing! Hyped to get this merged. 🫡

**Comment by technophile-04** [comment]
> hey @technophile-04 & @portdeveloper I've just released a new version to the @web3icons/react that reduces it's dependencies (mainly inquirer) as well as introduces some more network icons, thought I'd commit to the PR while it's still open.

Tysm @0xa3k5! For so quick updates!

**Review by technophile-04 — APPROVED** [review]

**Comment by technophile-04** [comment]
Just curious can't we use the same component for og images? Looking at https://github.com/BuidlGuidl/abi.ninja/blob/dd901afe23f482d54c7eade0974d120d3e53586a/packages/nextjs/pages/api/og.tsx#L210

It seems that we are using jsx here right?

**Comment by portdeveloper** [comment]
> Just curious can't we use the same component for og images? Looking at
> 
> https://github.com/BuidlGuidl/abi.ninja/blob/dd901afe23f482d54c7eade0974d120d3e53586a/packages/nextjs/pages/api/og.tsx#L210
> 
> It seems that we are using jsx here right?

I have just tested it and It does not work.


This could be well due to how image generation works in vercel/og, since it does not have access to the default nodejs runtime.

**Comment by 0xa3k5** [comment]
The <NetworkIcon /> essentially renders a `<svg>` or `` idk why it doesn't work, but I'm not very well educated about `vercel/og` is there a way for me to quickly test og generation? @portdeveloper

**Comment by portdeveloper** [comment]
> The `<NetworkIcon />` essentially renders a `<svg>` or `` idk why it doesn't work, but I'm not very well educated about `vercel/og` is there a way for me to quickly test og generation? @portdeveloper

Since you have abi.ninja on your computer I suggest that you go to:
image

Because this is just vercel/og, anything you put here should work:
image


You can do a `yarn start`

and check this route as an example:
[http://localhost:3000/api/og?contractAddress=0xde30da39c46104798bb5aa3fe8b9e0e1f348163f](http://localhost:3000/api/og?contractAddress=0xde30da39c46104798bb5aa3fe8b9e0e1f348163f)

**Comment by 0xa3k5** [comment]
oh tyvm @portdeveloper I'm seeing what the issue is: The `<NetworkIcon />` is a client-side component, it uses some hooks and dynamically imports the icons so the bundle size doesn't get bloated. The `vercel/og` however, works on the server so it can not resolve the NetworkIcon properyl. The `@web3icons/core` package serves svgs but it might be way overkill to add a dependency. I'm tinkering with a few ideas as a workaround that would not increase the dependencies. I'll let you know if I can figure out something

**Comment by portdeveloper** [comment]
@0xa3k5 Thanks for putting in the hard work and dissecting the issue!

**Pull request merged**